File: go/arrays/two_sum_test.go
package arrays

import (
	"testing"
)

func TestTwoSum(t *testing.T) {
	tests := []struct {
		name     string
		nums     []int
		target   int
		expected []int
	}{
		{
			name:     "basic case",
			nums:     []int{2, 7, 11, 15},
			target:   9,
			expected: []int{0, 1},
		},
		{
			name:     "different order",
			nums:     []int{3, 2, 4},
			target:   6,
			expected: []int{1, 2},
		},
		{
			name:     "no solution",
			nums:     []int{1, 2, 3},
			target:   10,
			expected: []int{},
		},
		{
			name:     "negative numbers",
			nums:     []int{-1, -2, -3, 5, 10},
			target:   7,
			expected: []int{3, 4},
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := TwoSum(tt.nums, tt.target)
			if !equal(result, tt.expected) && !equal(result, []int{tt.expected[1], tt.expected[0]}) {
				t.Errorf("got %v, want %v", result, tt.expected)
			}
		})
	}
}

func equal(a, b []int) bool {
	if len(a) != len(b) {
		return false
	}
	for i := range a {
		if a[i] != b[i] {
			return false
		}
	}
	return true
}

File: typescript/arrays/twoSum.test.ts
import { twoSum } from './twoSum';

describe('twoSum', () => {
  it('should find two numbers that sum to target', () => {
    const result = twoSum([2, 7, 11, 15], 9);
    expect(result.sort()).toEqual([0, 1]);
  });

  it('should handle different order', () => {
    const result = twoSum([3, 2, 4], 6);
    expect(result.sort()).toEqual([1, 2]);
  });

  it('should return empty array if no solution', () => {
    const result = twoSum([1, 2, 3], 10);
    expect(result).toEqual([]);
  });

  it('should handle negative numbers', () => {
    const result = twoSum([-1, -2, -3, 5, 10], 7);
    expect(result.sort()).toEqual([3, 4]);
  });
});
